2022/day7/day7.hs

```diff
@@ -0,0 +1,132 @@
+-- https://adventofcode.com/2022/day/7
+
+module Day7 where
+
+import Data.Function (on)
+import Data.List (find, intercalate, mapAccumL, minimumBy, sortBy)
+import Data.Map (fromListWith, toList)
+import Data.Maybe (fromMaybe)
+import System.Environment (getArgs)
+
+data DirectoryListing pathParts fileEntries = DirectoryListing [String] [([String], Int)]
```

Comment:
After reading this file, it seems to me like it contains three "sections:" data structure definitions, main, and parsing definitions.

If the order didn't matter to the Haskell compiler, I would order them with main first, then the parsing definitions, then the data structure definitions. (With big section comments above each one.) But maybe that's just my mainly-imperative brain. 🙂